soupsieve/css_match.py

```diff
@@ -956,9 +954,20 @@ def match_contains(self, el, contains):
                     content = self.get_text(el, no_iframe=self.is_html)
             found = False
             for text in contain_list.text:
-                if text in content:
-                    found = True
-                    break
+                if type(content) == str:
+                    if text in content:
+                        found = True
+                        break
+                elif type(content) == list:
```

Comment:
Sorry, I meant for the last comment here. We probably on need an else here as we only have two types of returns we can get here.

Reply:
I know. I tried to keep the function as general as possible :)